portfolio/src/main/webapp/gameoflife.js

```diff
@@ -0,0 +1,156 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// CODE FOR GAME OF LIFE
+
+// Page specific variables
+var canvas = document.getElementById("mycanvas");
+var ctx = canvas.getContext("2d");
+var stop = true;
+var interval, startTime, now, then, elapsed;
+
+resizeCanvasToDisplaySize(canvas);
+
+// GOL specific variables
+const SPEED = 200; // in miliseconds
+const CELL_SIZE = 10; // in pixels
+const SEED_SPAWN_RATE = 0.1; // as a ratio
+
+var rows = Math.floor(canvas.height / CELL_SIZE);
+var cols = Math.floor(canvas.width / CELL_SIZE);
+
+// Declare grid for GOL
+var grid = new Array(rows);
+for (var i = 0; i < grid.length; i++) {
+	grid[i] = new Array(cols);
+}
+
+
+// GOL specific helper functions
+function seedGrid() {
+    for (var x = 0; x < cols; x++) {
+        for (var y = 0; y < rows; y++) {
+            if (Math.random() < SEED_SPAWN_RATE) {
+                grid[y][x] = true;
+            }
+            else {
+                grid[y][x] = false;
+            }
+        }
+    } 
+}
+
+function updateGrid() {
```

Comment:
would be good to summarize the "game of life" in a comment above this func. Otherwise it's hard to understand the requisite business logic.